pkg/plugin/driver/base.go

```diff
@@ -211,3 +216,65 @@ func (b *Base) CreateWorkerContainerDefintion(hostname string, cert *tls.Certifi
 	}
 	return container
 }
+
+// DefaultPodSpec returns the default PodSpec used by DaemonSet plugins
+func defaultDaemonSetPodSpec() v1.PodSpec {
+	podSpec := v1.PodSpec{
+		Containers:         []v1.Container{},
+		DNSPolicy:          v1.DNSClusterFirstWithHostNet,
+		HostIPC:            true,
+		HostPID:            true,
+		HostNetwork:        true,
+		ServiceAccountName: "sonobuoy-serviceaccount",
+		Tolerations: []v1.Toleration{
+			{
+				Operator: v1.TolerationOpExists,
+			},
+		},
+		Volumes: []v1.Volume{
+			{
+				Name: "root",
+				VolumeSource: v1.VolumeSource{
+					HostPath: &v1.HostPathVolumeSource{
+						Path: "/",
+					},
+				},
+			},
+		},
+	}
+	return podSpec
+}
+
+// DefaultPodSpec returns the default PodSpec used by Job plugins
+func defaultJobPodSpec() v1.PodSpec {
+	return v1.PodSpec{
+		Containers:         []v1.Container{},
+		RestartPolicy:      v1.RestartPolicyNever,
+		ServiceAccountName: "sonobuoy-serviceaccount",
+		Tolerations: []v1.Toleration{
+			{
+				Key:      "node-role.kubernetes.io/master",
+				Operator: v1.TolerationOpExists,
+				Effect:   v1.TaintEffectNoSchedule,
+			},
+			{
+				Key:      "CriticalAddonsOnly",
+				Operator: v1.TolerationOpExists,
+			},
+		},
+		Volumes: []v1.Volume{},
+	}
+}
+
+// DefaultPodSpec returns the default pod spec used for the given plugin driver type.
+func DefaultPodSpec(d string) (v1.PodSpec, error) {
+	switch d {
```

Comment:
I believe I tried to avoid caring too much about the casing; on the server when dealing with plugins I just `strings.ToLower` the driver to ensure I don't fail someone for `daemonset` vs `Daemonset`